lib/philomena/tag_changes.ex

```diff
@@ -120,98 +256,48 @@ defmodule Philomena.TagChanges do
 
   """
   def delete_tag_change(%TagChange{} = tag_change) do
-    Repo.delete(tag_change)
+    tag_change
+    |> unindex_tag_change()
+    |> Repo.delete()
   end
 
   def count_tag_changes(field_name, value) do
     Repo.aggregate(from(tc in TagChange, where: field(tc, ^field_name) == ^value), :count, :id)
   end
 
-  def load(attrs, pagination) do
-    {tag_changes, _} = load(attrs, nil, pagination)
-
-    tag_changes
-  end
-
-  def load(attrs, count_field, pagination) do
-    query =
-      attrs
-      |> base_query()
-      |> added_or_tag_field(attrs)
-      |> filter_anon(attrs)
-
-    item_count =
-      if count_field do
-        Repo.one(from tc in query, select: count(field(tc, ^count_field), :distinct))
-      end
-
-    query =
-      query
-      |> preload([:user, image: [:user, :sources, tags: :aliases], tags: [:tag]])
-      |> group_by([tc], tc.id)
-      |> order_by(desc: :created_at)
-
-    {Repo.paginate(query, pagination), item_count}
-  end
-
-  defp base_query(%{ip: ip}) do
-    from tc in TagChange, where: fragment("? >>= ip", ^ip)
-  end
-
-  defp base_query(%{field: field_name, value: value}) do
-    from tc in TagChange, where: field(tc, ^field_name) == ^value
-  end
-
-  defp base_query(_) do
-    from(tc in TagChange)
+  def load(user, params, pagination) do
+    {:ok, query} = Query.compile(get_query(params), user: user)
+
+    TagChange
+    |> Search.search_definition(
+      %{
+        query: %{
+          bool: %{
+            must: [query]
+          }
+        },
+        sort: parse_sort(params)
+      },
+      pagination
+    )
+    |> Search.search_records(
+      preload(TagChange, [:user, image: [:user, :sources, tags: :aliases], tags: [:tag]])
+    )
   end
 
-  defp filter_anon(query, %{field: :user_id, value: id, filter_anon: true}) do
-    from t in query,
-      inner_join: i in Image,
-      on: i.id == t.image_id,
-      where: t.user_id == ^id and not (i.user_id == ^id and i.anonymous == true)
+  defp parse_sort(%{"tag_changes" => %{"sf" => sf, "sd" => sd}})
+       when sf in ["created_at", "tag_count", "added_tag_count", "removed_tag_count"] and
+              sd in ["desc", "asc"] do
+    %{sf => sd}
   end
 
-  defp filter_anon(query, _), do: query
-
-  defp added_or_tag_field(query, %{added: nil, tag: nil}), do: query
-
-  defp added_or_tag_field(query, attrs) do
-    query =
-      from tc in query,
-        inner_join: tct in TagChanges.Tag,
-        on: tc.id == tct.tag_change_id
-
-    query
-    |> added_field(attrs)
-    |> tag_field(attrs)
-    |> tag_id_field(attrs)
+  defp parse_sort(_params) do
+    %{created_at: :desc}
```

Comment:
(and other places) I think default sort should be ID. It wouldn't require a second sort (as per above)